packages/shared/ReactDOMFrameScheduling.js

```diff
@@ -146,10 +187,16 @@ if (!ExecutionEnvironment.canUseDOM) {
     }
   };
 
-  rIC = function(callback: (deadline: Deadline) => void): number {
+  rIC = function(
+    callback: (deadline: Deadline) => void,
+    options?: {timeout: number},
+  ): number {
     // This assumes that we only schedule one callback at a time because that's
     // how Fiber uses it.
     scheduledRICCallback = callback;
+    if (options != null && typeof options.timeout === 'number') {
```

Comment:
We also assume something about options being passed so you won't really need to test for it. Since this will be a required argument.